File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.py
class Solution:
    def distinctAverages(self, nums):
        st = set()
        nums.sort()
        n = len(nums)
        for i in range(n // 2):
            sum_ = nums[i] + nums[n - i - 1]
            st.add(sum_ / 2.0)
        return len(st)

File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.java
class Solution {
    public int distinctAverages(int[] nums) {
        Set<Double> st= new HashSet<>();
        Arrays.sort(nums);
        int n = nums.length;
        for(int i=0; i<n/2;i++){
            int sum = (nums[i]+ nums[n-i-1]);
            st.add((double)(sum)/2.0);
        }
        return st.size();
    }
}